File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/cpp_enum_runme.lua
require("import")	-- the import fn
import("cpp_enum")	-- import code
ce=cpp_enum    -- renaming import

-- catch "undefined" global variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})

assert(ce.ENUM_ONE ~= nil)
assert(ce.ENUM_TWO ~= nil)

-- Enums inside classes
assert(ce.Foo.Hi == 0)
assert(ce.Foo.Hello == 1);
-- old-style bindings
assert(ce.Foo_Hi == 0)
assert(ce.Foo_Hello == 1);

assert(ce.Hi == 0)
assert(ce.Hello == 1)

File: support_tools/swigwin-3.0.12/Examples/test-suite/javascript/cpp_enum_runme.js
var cpp_enum = require("cpp_enum");

var f = new cpp_enum.Foo()

if(f.hola != cpp_enum.Hello){
  print(f.hola);
  throw "Error";
}

f.hola = cpp_enum.Foo.Hi
if(f.hola != cpp_enum.Foo.Hi){
  print(f.hola);
  throw "Error";
}

f.hola = cpp_enum.Hello

if(f.hola != cpp_enum.Hello){
  print(f.hola);
  throw "Error";
}

cpp_enum.Foo.hi = cpp_enum.Hello
if(cpp_enum.Foo.hi != cpp_enum.Hello){
  print(cpp_enum.Foo.hi);
  throw "Error";
}

